packages/@magic-ext/oauth2/src/index.ts

```diff
@@ -64,6 +67,65 @@ export class OAuthExtension extends Extension.Internal<'oauth2'> {
 
     return getResult.call(this, queryString, lifespan);
   }
+
+  public loginWithPopup(configuration: OAuthPopupConfiguration) {
+    let popup: Window | null = null;
+
+    const promiEvent = createPromiEvent<OAuthPopupResult>(async (resolve, reject) => {
+      // see if the popup already exists.
+      if (popup) {
+        reject(new Error('Popup window already exists'));
+        return;
+      }
+
+      const width = 448;
+      const height = 568;
+      const left = window.screenLeft + (window.outerWidth / 2 - width / 2);
+      const top = window.screenTop + window.outerHeight * 0.15;
+      const jwt = await createJwt();
+
+      // try to open the pop up.
+      popup = window.open(
+        `https://auth.magic.link/oauth2/popup/start/${configuration.provider}?dpop=${jwt}&magicApiKey=${this.sdk.apiKey}`,
+        '_blank',
+        `width=${width},height=${height},left=${left},top=${top}`,
+      );
+
+      // see if the popup was blocked and throw an error accordingly
+      if (!popup) {
+        reject(new Error('Failed to open popup window'));
+        return;
+      }
+
+      const messageListener = (event: MessageEvent) => {
+        if (event.data.msgType !== MagicIncomingWindowMessage.MAGIC_POPUP_RESPONSE) return;
+
+        if (
+          event.data.method === MagicIncomingWindowMessage.MAGIC_POPUP_OAUTH_VERIFY_RESPONSE ||
+          event.data.method === MagicIncomingWindowMessage.MAGIC_POPUP_OAUTH_ERROR
+        ) {
+          window.removeEventListener('message', messageListener);
+
+          if (event.data.payload.error) {
+            reject(new Error(event.data.payload.error));
+          } else {
+            resolve(event.data.payload);
+          }
+        }
+      };
+
+      window.addEventListener('message', messageListener);
+
+      const checkPopupClosed = setInterval(() => {
+        if (popup?.closed) {
```

Comment:
I think there might be better way to check if the popup is closed. Could you check how passport sdk does this?

Reply:
I think this is the way we should check, passport also does it 
https://github.com/magiclabs/magic-passport/blob/cd21baf23ae60f7fa34639e34e425acb88e517cc/src/modules/base.ts#L56